File: server/app/models/db_version.py
"""
Database version tracking model.

Records each schema version applied to the database.
Used for upgrade detection and future automated migration support.
"""

from sqlalchemy import Column, Integer, String, DateTime, Text
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


class DBVersion(Base):
    """
    Database version tracking model.

    Each row represents a schema version that has been applied.
    The latest row indicates the current database schema version.
    """

    __tablename__ = "db_versions"

    id = Column(Integer, primary_key=True, autoincrement=True)
    version = Column(String(20), nullable=False)
    description = Column(Text, nullable=False, default='')
    applied_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, nullable=False)

File: server/internal/model/db_version.go
package model

import "time"

// DBVersion tracks database schema migration versions.
// Used to ensure the database schema is at the expected version.
type DBVersion struct {
	ID          int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Version     string    `gorm:"type:varchar(20);not null" json:"version"`
	Description string    `gorm:"type:text;not null;default:''" json:"description"`
	AppliedAt   time.Time `gorm:"not null;autoCreateTime;column:applied_at" json:"applied_at"`
}

func (DBVersion) TableName() string { return "db_versions" }
